data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetVersion.java

```diff
@@ -13,31 +13,53 @@
 
 package tech.pegasys.teku.beaconrestapi.handlers.v1.node;
 
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.CACHE_NONE;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_NODE;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.STRING_TYPE;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.javalin.core.util.Header;
 import io.javalin.http.Context;
-import io.javalin.http.Handler;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.function.Function;
 import org.jetbrains.annotations.NotNull;
-import tech.pegasys.teku.api.response.v1.node.Version;
 import tech.pegasys.teku.api.response.v1.node.VersionResponse;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
 import tech.pegasys.teku.infrastructure.version.VersionProvider;
-import tech.pegasys.teku.provider.JsonProvider;
 
-public class GetVersion implements Handler {
-  private final JsonProvider jsonProvider;
+public class GetVersion extends MigratingEndpointAdapter {
+  public static final String ROUTE = "/eth/v1/node/version";
 
-  public GetVersion(JsonProvider jsonProvider) {
-    this.jsonProvider = jsonProvider;
-  }
+  private static final SerializableTypeDefinition<VersionData> DATA_TYPE =
+      SerializableTypeDefinition.object(VersionData.class)
+          .withField("version", STRING_TYPE, VersionData::getVersion)
+          .build();
```

Comment:
@ajsutton this is probably a good example to ask about if we can define this serializable without creating a static object...

There's a number of similar examples of 'simple' objects that may have 1-2 fields that we're currently creating going to need to wrap with a static object in this way, and whether that's desirable...

Reply:
Yeah so if version is a string I'd just define something like:

```
SerializableTypeDefinition<String> DATA_TYPE = SerializableTypeDefinition.object(String.class)
  .withField("version", STRING_TYPE, Function.identity())
  .build();

SerializableTypeDefinition<String> RESPONSE_TYPE = SerializableTypeDefinition.object(String.class)
  .withField("data", DATA_TYPE, Function.identity())
  .build();
```

Which takes a single string and gives you all of:
```
{
  data: { version: "my version"; }
}
```